tests/kafkatest/services/kafka/kafka.py

```diff
@@ -91,12 +116,11 @@ def stop_node(self, node, clean_shutdown=True):
 
         for pid in pids:
             node.account.signal(pid, sig, allow_fail=False)
-
-        node.account.ssh("rm -f /mnt/kafka.pid", allow_fail=False)
+        wait_until(lambda: len(self.pids(node)) == 0, timeout_sec=20, err_msg="Kafka node failed to stop")
 
     def clean_node(self, node):
         node.account.kill_process("kafka", clean_shutdown=False, allow_fail=True)
-        node.account.ssh("rm -rf /mnt/kafka-logs /mnt/kafka.properties /mnt/kafka.log /mnt/kafka.pid", allow_fail=False)
+        node.account.ssh("rm -rf /mnt/kafka-logs /mnt/kafka.properties /mnt/kafka.log", allow_fail=False)
```

Comment:
Yup, I was thinking of failure modes as well... that's why even with services that use pid files, the cleanup code now always uses `ps` + `grep` :)

One specific "brittle" case I had in mind was something like this which was happening in the past:
- a call to `stop_node()` signals the process, but it fails to stop
- `stop_node()` call dutifully cleans up after itself and removes the pid file
- later, `clean_node` code finds no pid file, and therefore doesn't nuke the leftover process, causing later tests to fail

Granted, in this case `stop_node` should've verified the process went away before removing the pid file, but cleanup ideally should be resilient against such user error.

As for something like a "data namespace", yeah ultimately that would be better. And then we advance ever-so-incrementally toward providing ad-hoc service isolation, even on the same physical node